src/main/java/Parser.java

```diff
@@ -0,0 +1,216 @@
+     * @param tasks TaskList to hold tasks that the parser can interact with
+     * @param ui Ui to display results and errors to the user
+     * @return String representing the output from the UI to the user
+     */
+    public static String parse(String input, Storage storage, TaskList tasks, Ui ui) {
+        String output = ui.showDivider();
+
+        try {
+            if (input.equals("bye")) {
+                output += ui.showGoodbyeMessage();
+                done = true;
+            } else if (input.equals("list")) {
+                output += ui.showAllTasks(tasks);
+            } else if (input.startsWith("done")) {
+                output += markAsDone(input, tasks, ui);
+                storage.saveTasks(tasks, ui);
+            } else if (input.startsWith("delete")) {
+                output += deleteTask(input, tasks, ui);
+                storage.saveTasks(tasks, ui);
+            } else if (input.startsWith("find")) {
+                output += findTask(input, tasks, ui);
+            } else {
+
+                Task currTask = null;
+
+                if (input.startsWith("todo")) {
+                    currTask = createTodo(input);
+                } else if (input.startsWith("deadline")) {
+                    currTask = createDeadline(input);
+                } else if (input.startsWith("event")) {
+                    currTask = createEvent(input);
+                } else { // Invalid Command
+                    output += ui.showInvalidCommandMessage();
+                }
+
+                if (currTask != null) {
+                    output += addTask(currTask, tasks, ui);
+                    storage.saveTasks(tasks, ui);
+                }
+            }
+            return output + "\n" + ui.showDivider();
+        } catch (DukeException e) {
+            return ui.showError(e.getMessage());
+        }
+    }
+
+    private static String findTask(String input, TaskList tasks, Ui ui) throws DukeException {
+        if (input.equals("find")) {
+            throw new DukeException("Sorry sir you will have to try again and this time "
+                    + "tell me which task you want me to find");
+        }
+
+        String searchTerm = input.split(" ")[1];
+        TaskList searchResults = new TaskList();
+
+        for (int i = 0; i < tasks.getSize(); i++) {
+            Task task = tasks.getTask(i);
+            if (task.toString().contains(searchTerm)) {
+                searchResults.addTask(task);
+            }
+        }
+
+        return ui.showAllTasks(searchResults);
+    }
+
+    /**
+     * Marks the task provided by the user as completed.
+     *
+     * @param input User input that starts with the "done" keyword
+     * @param tasks TaskList that contains the task required
+     * @param ui Ui to display results and errors to the user
+     * @return String representing the output from the UI to the user
+     * @throws DukeException Thrown when the user does not specify a task
```

Comment:
Can remove the word "Thrown" here and in other instances? I think '@throws DukeException when ....' is meant to be read like a full sentence